src/beans.sol

```diff
@@ -0,0 +1,176 @@
+// Copyright (C) 2018  Rain <rainbreak@riseup.net>, lucasvo
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "ds-note/note.sol";
+
+// Beans
+// Keeps track of interest rate accumulators (chi values) for all interest rate categories.
+// Calculates debt each loan according to its interest rate category and pie value.
+contract Beans is DSNote {
+    // --- Auth ---
+    mapping (address => uint) public wards;
+    function rely(address usr) public auth note { wards[usr] = 1; }
+    function deny(address usr) public auth note { wards[usr] = 0; }
+    modifier auth { require(wards[msg.sender] == 1); _; }
+    
+    // --- Data ---
+    // https://github.com/makerdao/dsr/blob/master/src/dsr.sol
+    struct Fee {
+        uint debt;
+        uint chi;
+        uint speed; // Accumulation per second
+        uint48 rho; // Last time the rate was accumulated
+    }
+
+    mapping (uint => Fee) public fees;
+    mapping (uint => uint) public pie;
+
+    uint public debt;
```

Comment:
I think we should only call the loan `debt`  => debt and refer fee and global debt as `totalDebt`.

Otherwise `incDebt` method names are confusing because we don't know which debt we increase